src/gt4py/cartesian/frontend/gtscript_frontend.py

```diff
@@ -999,7 +1001,12 @@ def visit_Constant(
                 loc=nodes.Location.from_ast_node(node),
             )
         elif isinstance(value, numbers.Number):
-            data_type = nodes.DataType.from_dtype(np.dtype(type(value)))
+            value_type = (
+                self.dtypes[type(value)]
+                if self.dtypes and type(value) in self.dtypes.keys()
+                else np.dtype(type(value))
+            )
```

Comment:
There should be a check that raises if keys other than `int` or `float` are used.

Reply:
raise GTScriptSyntaxError

Reply:
Actually no, you can do `FloatField["my_string"]` then `dtypes={"my_string":float}`, so not a raise